### PR TITLE
API and build changes for new SYCL compiler on corona.

### PR DESCRIPTION
**Pull request by rchen20**
# Summary

- This PR is an update of the SYCL interface and compiler
- It does the following:
  - Allows usage of an updated SYCL compiler on Corona (https://github.com/LLNL/RAJA/issues/1255)
  - Modifies SYCL build scripts
  - Changes some SYCL API calls
  - Changes Desul SYCL API calls

### CODE REVIEW

File: include/RAJA/policy/sycl/reduce.hpp [hunk 1]
```diff
@@ -251,7 +251,7 @@ struct TargetReduce
   {
 #ifdef __SYCL_DEVICE_ONLY__
     auto i = 0; //__spirv::initLocalInvocationId<1, cl::sycl::id<1>>()[0];
-    auto atm = cl::sycl::ext::oneapi::atomic_ref<T, cl::sycl::memory_order_acq_rel, cl::sycl::memory_scope::device, cl::sycl::access::address_space::global_space>(val.device[i]);
+    auto atm = ::sycl::atomic_ref<T, cl::sycl::memory_order_acq_rel, cl::sycl::memory_scope::device, cl::sycl::access::address_space::global_space>(val.device[i]);
```
Comment (rchen20) [inline include/RAJA/policy/sycl/reduce.hpp:254]: Hi @homerdin, I've built a new SYCL compiler from scratch on LLNL's Corona machine (compiler build process here https://github.com/LLNL/RAJA/issues/1255), and ran in to some SYCL API problems that were solved by changing some namespaces. Are these changes ok with you?

**Comment by rchen20** [comment]
Hi @trws @rhornung67, in order to get this new SYCL compiler to work, I had to change some namespaces in this Desul file https://github.com/desul/desul/blob/master/atomics/include/desul/atomics/Adapt_SYCL.hpp. It appears that the latest version of that file is correct, but our RAJA version of Desul is out-of-date. Should I update Desul in RAJA to accommodate these changes?

**Comment by rhornung67** [comment]
@rchen20 there is a PR to update desul in RAJA, but it is on hold pending a BLT update https://github.com/LLNL/RAJA/pull/1563  The need for an update for a local compiler make the desul update more urgent. I will push on the BLT update.

**Comment by rchen20** [comment]
@rhornung67 @homerdin These changes should be ready to go. Please review when you have a spare moment. Thanks!

**Review by homerdin — APPROVED** [review]

**Review by rhornung67 — APPROVED** [review]

**Comment by rhornung67** [comment]
@rchen20 are you good with this? I think it can be merged.

**Comment by rchen20** [comment]
> @rchen20 are you good with this? I think it can be merged.

Yes, I'll merge this. Thanks for trying it out.

**Pull request merged**